packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart

```diff
@@ -204,6 +205,209 @@ class _FieldSizeColumn extends ColumnData<ProfileRecord> {
+  String getDisplayValue(AdaptedProfile dataObject) {
+    return prettyPrintBytes(getValue(dataObject), includeUnit: true)!;
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class _HeapCapacityColumn extends _HeapStatsColumn {
+  _HeapCapacityColumn({required super.generation})
+      : super(
+          'Capacity',
+          titleTooltip:
+              'The current size of the heap, including unallocated memory',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  int? getValue(AdaptedProfile dataObject) {
+    return getGCStats(dataObject).capacity;
+  }
+
+  @override
+  String getDisplayValue(AdaptedProfile dataObject) {
+    return prettyPrintBytes(getValue(dataObject), includeUnit: true)!;
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class _HeapCollectionsColumn extends _HeapStatsColumn {
+  _HeapCollectionsColumn({required super.generation})
+      : super(
+          'Collections',
+          titleTooltip: 'The number of garbage collections run on the heap',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  int? getValue(AdaptedProfile dataObject) {
+    return getGCStats(dataObject).collections;
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class _HeapCollectionsTime extends _HeapStatsColumn {
+  _HeapCollectionsTime({required super.generation})
+      : super(
+          'Latency',
+          titleTooltip:
+              'The average time taken to perform a garbage collection on the heap (ms)',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  double? getValue(AdaptedProfile dataObject) {
+    final stats = getGCStats(dataObject);
+    if (stats.averageCollectionTime.isNaN) {
+      return 0;
+    }
+    return getGCStats(dataObject).averageCollectionTime * 1000;
+  }
+
+  @override
+  String getDisplayValue(AdaptedProfile dataObject) {
+    return msText(
+      Duration(
+        microseconds: getValue(dataObject)!.toInt(),
+      ),
+      fractionDigits: 2,
+    );
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class _GCStatsTable extends StatelessWidget {
+  const _GCStatsTable({
+    Key? key,
+    required this.controller,
+  }) : super(key: key);
+
+  static final _columnGroup = [
+    ColumnGroup(
+      title: '',
+      range: const Range(0, 1),
+    ),
+    ColumnGroup(
+      title: 'Total',
+      range: const Range(1, 5),
+    ),
+    ColumnGroup(
+      title: 'New Space',
+      range: const Range(5, 9),
+    ),
+    ColumnGroup(
+      title: 'Old Space',
+      range: const Range(9, 13),
+    ),
+  ];
+
+  static final _columns = [
+    _HeapNameColumn(),
+    for (final generation in [
+      _HeapGeneration.total,
+      _HeapGeneration.newSpace,
+      _HeapGeneration.oldSpace,
+    ]) ...[
```

Comment:
loop through `_HeapGeneration.values` here instead

Reply:
Ordering is important here, which is why I decided to iterate through this list instead.